### PR TITLE
Add drilled tile rebased

### PR DESCRIPTION
**Pull request by mkofler96**
# Overview
added new microtile

## Addressed issues
.

## Showcase
A "short" example to highlight the (new) feature
```
# %%
import splinepy as sp
import gustaf as gus
import numpy as np
import vedo

# comment this line to visualize within the notebook (limited functionalities)
vedo.settings.default_backend = "vtk"

# %%
microstructure = sp.microstructure.Microstructure()
microstructure.deformation_function = sp.Bezier(
    degrees=[1, 1],
    control_points=[[0, 0], [0.5,0], [0, 1], [0.5, 1]],
)

angles = np.linspace(-np.pi/2+0.01,np.pi/2-0.01, 10)

tiles = []
for a in angles:
    parameters = np.ones((1,1))*a
    my_microtile = sp.microstructure.tiles.DrilledCrossTile1D()
    showable_tile = my_microtile.create_tile(parameters=parameters)
    tiles.append(showable_tile[0])
gus.show(*tiles, control_points=False)

# %%
# Parameter spline
para_s = sp.BSpline(
    degrees=[1, 1],
    knot_vectors=[[0, 0, 1, 1], [0, 0,0.5, 1, 1]],
    control_points=[[1], [0], [0], [0] , [1], [0]],
)


def pf_curved_tile(x):
    """
    Parametrization Function (determines thickness)
    """
    
    return np.pi/2*para_s.evaluate(x)

# %%
ms = sp.microstructure.Microstructure()
ms.deformation_function = sp.Bezier(
    degrees=[1, 1],
    control_points=[[0, 0], [0.5,0], [0, 1], [0.5, 1]],
)

my_microtile = sp.microstructure.tiles.DrilledCrossTile1D()
showable_tile = my_microtile.create_tile()

#ms.microtile = showable_tile[0]
ms.microtile = my_microtile
ms.tiling = [8, 16]
ms.parametrization_function = pf_curved_tile
ms.show(control_points=False, title="2D Lattice Microstructure", lighting="off")



```

## Checklists
* [x] Documentations are up-to-date.
* [ ] Added example(s)
* [ ] Added test(s)

### CODE REVIEW

**Review by jzwar — APPROVED** [review]
LGTM, thanks for the update!

File: splinepy/microstructure/tiles/drilled_cross_beams.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
+import numpy as _np
+
+from splinepy.bezier import Bezier as _Bezier
+from splinepy.microstructure.tiles.tilebase import TileBase as _TileBase
+
+
+class DrilledCrossBeams(_TileBase):
+    def __init__(self):
+        """DrilledCrossBeams"""
+        self._dim = 2
```
Comment (j042) [inline splinepy/microstructure/tiles/drilled_cross_beams.py:10]: _para_dim is missing here

**Review by j042 — CHANGES_REQUESTED** [review]

**Comment by mkofler96** [comment]
First it was called DrilledCrossTile1D, but Jacques suggested to give it another name:

The name of the file should be in `snake_case`, please. Also, consider naming the tile
`DrilledCrossBeams` or `TwistedCrossBeams` to avoid confusion...

_Originally posted by @jzwar in https://github.com/tataratat/splinepy/issues/294#issuecomment-1810012860_

**Comment by j042** [comment]
@

> First it was called DrilledCrossTile1D, but Jacques suggested to give it another name:
> 
> The name of the file should be in `snake_case`, please. Also, consider naming the tile `DrilledCrossBeams` or `TwistedCrossBeams` to avoid confusion...
> 
> _Originally posted by @jzwar in [#294 (comment)](https://github.com/tataratat/splinepy/pull/294#issuecomment-1810012860)_

so, `Cross1D`? @jzwar

**Comment by j042** [comment]
@jzwar don't we already have `Cross2D`/`Cross3D`? However, I guess this is `X`. What about `X`?

**Comment by j042** [comment]
as `X` is not recommended class name, let's go with `Chi`

**Review by j042 — APPROVED** [review]
Thanks!

**Pull request merged**